--- a/feeds/packages/multimedia/ffmpeg/Makefile	2017-11-28 13:11:13.226666666 +0000
+++ b/feeds/packages/multimedia/ffmpeg/Makefile	2017-11-28 13:12:16.000000000 +0000
@@ -294,8 +294,11 @@
 $(call Package/libffmpeg/Default)
  TITLE+= (custom)
  DEPENDS+= +FFMPEG_CUSTOM_SELECT_libopus:libopus \
-           +PACKAGE_libx264:libx264 +PACKAGE_lame-lib:lame-lib
+           +PACKAGE_libx264:libx264
 
+ifeq ($(CONFIG_FFMPEG_CUSTOM_SELECT_mp3lame),y)
+ DEPENDS+=+lame-lib
+endif
  VARIANT:=custom
  MENU:=1
 endef
