diff -Naur a/metadata.c b/metadata.c
--- a/metadata.c	2013-11-02 08:44:41.000000000 +0800
+++ b/metadata.c	2016-10-09 09:00:57.537276539 +0800
@@ -757,10 +757,10 @@
 		aac_object_type_t aac_type = AAC_INVALID;
 		switch( ac->codec_id )
 		{
-			case CODEC_ID_MP3:
+			case AV_CODEC_ID_MP3:
 				audio_profile = PROFILE_AUDIO_MP3;
 				break;
-			case CODEC_ID_AAC:
+			case AV_CODEC_ID_AAC:
 				if( !ac->extradata_size ||
 				    !ac->extradata )
 				{
@@ -801,12 +801,12 @@
 						break;
 				}
 				break;
-			case CODEC_ID_AC3:
-			case CODEC_ID_DTS:
+			case AV_CODEC_ID_AC3:
+			case AV_CODEC_ID_DTS:
 				audio_profile = PROFILE_AUDIO_AC3;
 				break;
-			case CODEC_ID_WMAV1:
-			case CODEC_ID_WMAV2:
+			case AV_CODEC_ID_WMAV1:
+			case AV_CODEC_ID_WMAV2:
 				/* WMA Baseline: stereo, up to 48 KHz, up to 192,999 bps */
 				if ( ac->bit_rate <= 193000 )
 					audio_profile = PROFILE_AUDIO_WMA_BASE;
@@ -815,19 +815,19 @@
 					audio_profile = PROFILE_AUDIO_WMA_FULL;
 				break;
 			#if LIBAVCODEC_VERSION_INT > ((51<<16)+(50<<8)+1)
-			case CODEC_ID_WMAPRO:
+			case AV_CODEC_ID_WMAPRO:
 				audio_profile = PROFILE_AUDIO_WMA_PRO;
 				break;
 			#endif
-			case CODEC_ID_MP2:
+			case AV_CODEC_ID_MP2:
 				audio_profile = PROFILE_AUDIO_MP2;
 				break;
-			case CODEC_ID_AMR_NB:
+			case AV_CODEC_ID_AMR_NB:
 				audio_profile = PROFILE_AUDIO_AMR;
 				break;
 			default:
-				if( (ac->codec_id >= CODEC_ID_PCM_S16LE) &&
-				    (ac->codec_id < CODEC_ID_ADPCM_IMA_QT) )
+				if( (ac->codec_id >= AV_CODEC_ID_PCM_S16LE) &&
+				    (ac->codec_id < AV_CODEC_ID_ADPCM_IMA_QT) )
 					audio_profile = PROFILE_AUDIO_PCM;
 				else
 					DPRINTF(E_DEBUG, L_METADATA, "Unhandled audio codec [0x%X]\n", ac->codec_id);
@@ -864,7 +864,7 @@
 		if( strcmp(ctx->iformat->name, "avi") == 0 )
 		{
 			xasprintf(&m.mime, "video/x-msvideo");
-			if( vc->codec_id == CODEC_ID_MPEG4 )
+			if( vc->codec_id == AV_CODEC_ID_MPEG4 )
 			{
         			fourcc[0] = vc->codec_tag     & 0xff;
 			        fourcc[1] = vc->codec_tag>>8  & 0xff;
@@ -888,7 +888,7 @@
 
 		switch( vc->codec_id )
 		{
-			case CODEC_ID_MPEG1VIDEO:
+			case AV_CODEC_ID_MPEG1VIDEO:
 				if( strcmp(ctx->iformat->name, "mpeg") == 0 )
 				{
 					if( (vc->width  == 352) &&
@@ -899,7 +899,7 @@
 					xasprintf(&m.mime, "video/mpeg");
 				}
 				break;
-			case CODEC_ID_MPEG2VIDEO:
+			case AV_CODEC_ID_MPEG2VIDEO:
 				m.dlna_pn = malloc(64);
 				off = sprintf(m.dlna_pn, "MPEG_");
 				if( strcmp(ctx->iformat->name, "mpegts") == 0 )
@@ -972,7 +972,7 @@
 					m.dlna_pn = NULL;
 				}
 				break;
-			case CODEC_ID_H264:
+			case AV_CODEC_ID_H264:
 				m.dlna_pn = malloc(128);
 				off = sprintf(m.dlna_pn, "AVC_");
 
@@ -990,11 +990,11 @@
 						          vc->height * vc->sample_aspect_ratio.den,
 						          1024*1024);
 					}
-					if (ctx->streams[video_stream]->r_frame_rate.den)
-						fps = ctx->streams[video_stream]->r_frame_rate.num / ctx->streams[video_stream]->r_frame_rate.den;
+					if (ctx->streams[video_stream]->avg_frame_rate.den)
+						fps = ctx->streams[video_stream]->avg_frame_rate.num / ctx->streams[video_stream]->avg_frame_rate.den;
 					else
 						fps = 0;
-					interlaced = vc->time_base.den ? (ctx->streams[video_stream]->r_frame_rate.num / vc->time_base.den) : 0;
+					interlaced = vc->time_base.den ? (ctx->streams[video_stream]->avg_frame_rate.num / vc->time_base.den) : 0;
 					if( ((((vc->width == 1920 || vc->width == 1440) && vc->height == 1080) ||
 					      (vc->width == 720 && vc->height == 480)) && fps == 59 && interlaced) ||
 					    ((vc->width == 1280 && vc->height == 720) && fps == 59 && !interlaced) )
@@ -1283,7 +1283,7 @@
 				}
 				DPRINTF(E_DEBUG, L_METADATA, "Stream %d of %s is h.264\n", video_stream, basepath);
 				break;
-			case CODEC_ID_MPEG4:
+			case AV_CODEC_ID_MPEG4:
         			fourcc[0] = vc->codec_tag     & 0xff;
 			        fourcc[1] = vc->codec_tag>>8  & 0xff;
 			        fourcc[2] = vc->codec_tag>>16 & 0xff;
@@ -1346,7 +1346,7 @@
 					}
 				}
 				break;
-			case CODEC_ID_WMV3:
+			case AV_CODEC_ID_WMV3:
 				/* I'm not 100% sure this is correct, but it works on everything I could get my hands on */
 				if( vc->extradata_size > 0 )
 				{
@@ -1355,7 +1355,7 @@
 					if( !((vc->extradata[0] >> 6) & 1) )
 						vc->profile = 0;
 				}
-			case CODEC_ID_VC1:
+			case AV_CODEC_ID_VC1:
 				if( strcmp(ctx->iformat->name, "asf") != 0 )
 				{
 					DPRINTF(E_DEBUG, L_METADATA, "Skipping DLNA parsing for non-ASF VC1 file %s\n", path);
@@ -1454,7 +1454,7 @@
 					}
 				}
 				break;
-			case CODEC_ID_MSMPEG4V3:
+			case AV_CODEC_ID_MSMPEG4V3:
 				xasprintf(&m.mime, "video/x-msvideo");
 			default:
 				DPRINTF(E_DEBUG, L_METADATA, "Stream %d of %s is %s [type %d]\n",
