diff -Naur c/block.c d/block.c
--- c/block.c	2016-11-02 16:57:25.830772485 +0800
+++ d/block.c	2016-11-02 17:01:00.887356118 +0800
@@ -568,7 +568,7 @@
 	_cache_load("/dev/mapper/*");
 }
 
-
+/*
 static int print_block_uci(struct probe_info *pr)
 {
 	if (!strcmp(pr->type, "swap")) {
@@ -586,7 +586,7 @@
 
 	return 0;
 }
-
+*/
 static struct probe_info* find_block_info(char *uuid, char *label, char *path)
 {
 	struct probe_info *pr = NULL;
@@ -1543,7 +1543,7 @@
 
 static int main_detect(int argc, char **argv)
 {
-	struct probe_info *pr;
+	/*struct probe_info *pr;*/
 
 	cache_load(0);
 	printf("config 'global'\n");
@@ -1553,9 +1553,9 @@
 	printf("\toption\tauto_mount\t'1'\n");
 	printf("\toption\tdelay_root\t'5'\n");
 	printf("\toption\tcheck_fs\t'0'\n\n");
-	list_for_each_entry(pr, &devices, list)
+	/*list_for_each_entry(pr, &devices, list)
 		print_block_uci(pr);
-
+*/
 	return 0;
 }
 
