--- a/package/kernel/linux/modules/netsupport.mk	2016-10-05 21:20:07.000000000 +0800
+++ b/package/kernel/linux/modules/netsupport.mk	2016-10-31 12:55:13.496479732 +0800
@@ -734,7 +734,7 @@
 
 SCHED_MODULES = $(patsubst $(LINUX_DIR)/net/sched/%.ko,%,$(wildcard $(LINUX_DIR)/net/sched/*.ko))
 SCHED_MODULES_CORE = sch_ingress sch_fq_codel sch_hfsc cls_fw cls_route cls_flow cls_tcindex cls_u32 em_u32 act_mirred act_skbedit
-SCHED_MODULES_FILTER = $(SCHED_MODULES_CORE) act_connmark sch_netem
+SCHED_MODULES_FILTER = $(SCHED_MODULES_CORE) act_connmark sch_netem sch_esfq
 SCHED_MODULES_EXTRA = $(filter-out $(SCHED_MODULES_FILTER),$(SCHED_MODULES))
 SCHED_FILES = $(patsubst %,$(LINUX_DIR)/net/sched/%.ko,$(filter $(SCHED_MODULES_CORE),$(SCHED_MODULES)))
 SCHED_FILES_EXTRA = $(patsubst %,$(LINUX_DIR)/net/sched/%.ko,$(SCHED_MODULES_EXTRA))
@@ -813,6 +813,18 @@
 
 $(eval $(call KernelPackage,sched))
 
+define KernelPackage/sched-esfq
+  SUBMENU:=$(NETWORK_SUPPORT_MENU)
+  TITLE:=Traffic shaper ESFQ support
+  KCONFIG:= \
+	CONFIG_NET_SCH_ESFQ \
+	CONFIG_NET_SCH_ESFQ_NFCT=y
+  DEPENDS:=+kmod-sched-core +kmod-ipt-core +kmod-ipt-conntrack
+  FILES:=$(LINUX_DIR)/net/sched/sch_esfq.ko
+  AUTOLOAD:=$(call AutoLoad,73, sch_esfq)
+endef
+$(eval $(call KernelPackage,sched-esfq))
+
 
 define KernelPackage/ax25
   SUBMENU:=$(NETWORK_SUPPORT_MENU)
