--- a/Makefile	2016-10-10 16:44:04.000000000 +0800
+++ b/Makefile	2016-10-10 16:44:13.386307272 +0800
@@ -115,7 +115,6 @@
 	--disable-dts \
 	--disable-resindvd \
 	--disable-faac \
-	--disable-faad \
 	--disable-flite \
 	--disable-gsm \
 	--disable-kate \
@@ -251,6 +250,7 @@
 $(eval $(call GstBuildPlugin,dvdspu,dvdspu support,video,,))
 $(eval $(call GstBuildPlugin,fbdevsink,fbdev support,video,,))
 $(eval $(call GstBuildPlugin,festival,festival support,audio,,))
+$(eval $(call GstBuildPlugin,faad,faad support,audio,,+libfaad2))
 $(eval $(call GstBuildPlugin,hls,HLS support,pbutils adaptivedemux,multifile,+libnettle))
 $(eval $(call GstBuildPlugin,frei0r,frei0r support,controller video,,))
 $(eval $(call GstBuildPlugin,id3tag,id3tag support,tag,,))
