--- a/feeds/luci/applications/luci-app-samba/luasrc/model/cbi/samba.lua	2016-09-20 21:13:00.523965910 +0800
+++ b/feeds/luci/applications/luci-app-samba/luasrc/model/cbi/samba.lua	2016-09-21 09:12:00.369314776 +0800
@@ -13,7 +13,7 @@
 s:taboption("general", Value, "name", translate("Hostname"))
 s:taboption("general", Value, "description", translate("Description"))
 s:taboption("general", Value, "workgroup", translate("Workgroup"))
-s:taboption("general", Value, "homes", translate("Share home-directories"),
+s:taboption("general", Flag, "homes", translate("Share home-directories"),
         translate("Allow system users to reach their home directories via " ..
                 "network shares"))
 
