diff -Naur openwrt/package/feeds/packages/sqm-scripts/Makefile openwrt.new/package/feeds/packages/sqm-scripts/Makefile
--- package/feeds/packages/sqm-scripts/Makefile	2015-06-21 08:25:49.267034700 +0200
+++ package/feeds/packages/sqm-scripts/Makefile	2015-06-21 08:14:53.000000000 +0200
@@ -19,7 +19,7 @@
 define Package/sqm-scripts
   SECTION:=net
   CATEGORY:=Base system
-  DEPENDS:=@(PACKAGE_tc||PACKAGE_tc-adv) +kmod-sched +kmod-ifb iptables +ip \
+  DEPENDS:=@(PACKAGE_tc||PACKAGE_tc-adv) +kmod-sched +kmod-ifb iptables +!PACKAGE_ip-full:ip \
	+iptables-mod-ipopt +iptables-mod-conntrack-extra
  TITLE:=SQM Scripts (QoS)
  PKGARCH:=all
