diff -Naur openwrt-old/feeds/luci/applications/luci-app-minidlna/luasrc/model/cbi/minidlna.lua openwrt/feeds/luci/applications/luci-app-minidlna/luasrc/model/cbi/minidlna.lua
--- openwrt-old/feeds/luci/applications/luci-app-minidlna/luasrc/model/cbi/minidlna.lua	2015-06-05 11:40:58.624020713 +0800
+++ openwrt/feeds/luci/applications/luci-app-minidlna/luasrc/model/cbi/minidlna.lua	2015-06-05 11:51:00.000000000 +0800
@@ -84,6 +84,11 @@
 o.rmempty = true
 o.placeholder = "/var/log"
 
+o = s:taboption("advanced", Value, "locale", translate("Locale:"),
+	translate("Set locale when MiniDLNA starts."))
+o.rmempty = true
+o.placeholder = "en_US.UTF-8"
+
 s:taboption("advanced", Flag, "inotify", translate("Enable inotify:"),
 	translate("Set this to enable inotify monitoring to automatically discover new files."))
 
diff -Naur openwrt-old/feeds/packages/multimedia/minidlna/files/minidlna.config openwrt/feeds/packages/multimedia/minidlna/files/minidlna.config
--- openwrt-old/feeds/packages/multimedia/minidlna/files/minidlna.config	2015-06-05 11:40:59.139996810 +0800
+++ openwrt/feeds/packages/multimedia/minidlna/files/minidlna.config	2015-06-05 11:49:24.000000000 +0800
@@ -15,3 +15,4 @@
 	option root_container '.'
 	list media_dir '/mnt'
 	option album_art_names 'Cover.jpg/cover.jpg/AlbumArtSmall.jpg/albumartsmall.jpg/AlbumArt.jpg/albumart.jpg/Album.jpg/album.jpg/Folder.jpg/folder.jpg/Thumb.jpg/thumb.jpg'
+	option locale 'zh_CN.UTF-8'
diff -Naur openwrt-old/feeds/packages/multimedia/minidlna/files/minidlna.init openwrt/feeds/packages/multimedia/minidlna/files/minidlna.init
--- openwrt-old/feeds/packages/multimedia/minidlna/files/minidlna.init	2015-06-05 11:40:59.139996810 +0800
+++ openwrt/feeds/packages/multimedia/minidlna/files/minidlna.init	2015-06-05 11:48:53.000000000 +0800
@@ -46,6 +46,7 @@
 
 	config_get port $cfg port
 	config_get interface $cfg interface
+	config_get locale $1 locale
 
 	[ -z "$interface" -o -t "$port" ] && return 1
 
@@ -89,6 +90,7 @@
 
 	mkdir -m 0755 -p $db_dir
 	mkdir -m 0755 -p $log_dir
+	[[ -z "$locale" ]] || export LANG=$locale
 	service_start /usr/bin/minidlna -f "$MINIDLNA_CONFIG_FILE"
 }
 
--- openwrt-old/feeds/luci/applications/luci-app-minidlna/po/zh-cn/minidlna.po	2015-06-08 14:20:37.000000000 +0200
+++ openwrt/feeds/luci/applications/luci-app-minidlna/po/zh-cn/minidlna.po	2015-06-10 12:38:07.422794500 +0200
@@ -170,3 +170,9 @@
 
 msgid "miniSSDP socket:"
 msgstr "miniSSDP socket："
+
+msgid "Set locale when MiniDLNA starts."
+msgstr "设置语言环境"
+
+msgid "Locale:"
+msgstr "语言环境："
