diff -Naur a/package/feeds/packages/memcached/Makefile b/package/feeds/packages/memcached/Makefile
--- a/package/feeds/packages/memcached/Makefile	2015-07-16 19:23:31.000000000 +0200
+++ b/package/feeds/packages/memcached/Makefile	2015-07-17 10:49:21.975093800 +0200
@@ -24,7 +24,7 @@
 define Package/memcached
   SECTION:=net
   CATEGORY:=Network
-  DEPENDS:=+libevent +libpthread
+  DEPENDS:=+libevent2 +libpthread
   TITLE:=The high-performance, distributed memory object caching system
   URL:=http://memcached.org/
 endef
diff -Naur a/package/feeds/packages/nfs-kernel-server/Makefile b/package/feeds/packages/nfs-kernel-server/Makefile
--- a/package/feeds/packages/nfs-kernel-server/Makefile	2015-07-16 19:23:31.000000000 +0200
+++ b/package/feeds/packages/nfs-kernel-server/Makefile	2015-07-17 10:49:33.765606700 +0200
@@ -60,7 +60,7 @@
   $(call Package/nfs-kernel-server/Default)
   SECTION:=utils
   CATEGORY:=Utilities
-  DEPENDS+= +libevent +USE_UCLIBC:librpc
+  DEPENDS+= +libevent2 +USE_UCLIBC:librpc
   TITLE:=updated mount utility (includes nfs4)
 endef
 
