diff -Naur a/package/feeds/luci/luci-app-p2pblock/root/etc/uci-defaults/luci-p2pblock b/package/feeds/luci/luci-app-p2pblock/root/etc/uci-defaults/luci-p2pblock
--- a/package/feeds/luci/luci-app-p2pblock/root/etc/uci-defaults/luci-p2pblock	2015-06-29 09:32:26.000000000 +0200
+++ b/package/feeds/luci/luci-app-p2pblock/root/etc/uci-defaults/luci-p2pblock	2015-07-04 14:45:43.000000000 +0200
@@ -7,5 +7,7 @@
 	commit ucitrack
 EOF
 
+/etc/init.d/freifunk-p2pblock disable
+
 rm -f /tmp/luci-indexcache
 exit 0
