--- a/feeds/packages/utils/watchcat/Makefile	2016-10-05 21:25:20.000000000 +0800
+++ b/feeds/packages/utils/watchcat/Makefile	2016-11-13 19:57:40.773634506 +0800
@@ -38,8 +38,6 @@
 	$(INSTALL_BIN) ./files/initd_watchcat $(1)/etc/init.d/watchcat
 	$(INSTALL_DIR) $(1)/usr/bin
 	$(INSTALL_BIN) ./files/watchcat.sh $(1)/usr/bin/watchcat.sh
-	$(INSTALL_DIR) $(1)/etc/uci-defaults
-	$(INSTALL_BIN) ./files/uci_defaults_watchcat $(1)/etc/uci-defaults/50-watchcat
 endef
 
 $(eval $(call BuildPackage,watchcat))
