diff -Naur a/block.c b/block.c
--- a/block.c	2016-10-31 23:36:17.000000000 +0800
+++ b/block.c	2016-11-02 16:50:27.650350965 +0800
@@ -241,6 +241,55 @@
 	free(optstr);
 }
 
+static const char *mount_flags_to_string(unsigned long mountflags)
+{
+	static char param[256];
+
+	param[0] = 0;
+
+	if (mountflags & MS_SYNCHRONOUS)
+		strcat(param, "sync,");
+	else
+		strcat(param, "async,");
+
+	if (mountflags & MS_DIRSYNC)
+		strcat(param, "dirsync,");
+
+	if (mountflags & MS_MANDLOCK)
+		strcat(param, "mand,");
+
+	if (mountflags & MS_NOATIME)
+		strcat(param, "noatime,");
+
+	if (mountflags & MS_NODEV)
+		strcat(param, "nodev,");
+
+	if (mountflags & MS_NODIRATIME)
+		strcat(param, "nodiratime,");
+
+	if (mountflags & MS_NOEXEC)
+		strcat(param, "noexec,");
+
+	if (mountflags & MS_NOSUID)
+		strcat(param, "nosuid,");
+
+	if (mountflags & MS_RDONLY)
+		strcat(param, "ro,");
+	else
+		strcat(param, "rw,");
+
+	if (mountflags & MS_RELATIME)
+		strcat(param, "relatime,");
+
+	if (mountflags & MS_STRICTATIME)
+		strcat(param, "strictatime,");
+
+	if (param[0])
+		param[strlen(param) - 1] = 0;
+
+	return param;
+}
+
 static int mount_add(struct uci_section *s)
 {
 	struct blob_attr *tb[__MOUNT_MAX] = { 0 };
@@ -530,6 +579,7 @@
 		printf("\toption\tuuid\t'%s'\n", pr->uuid);
 	else
 		printf("\toption\tdevice\t'%s'\n", pr->dev);
+	printf("\toption\tfstype\t'%s'\n", (!strcmp(pr->type, "ntfs") ? "ntfs-3g" : pr->type));
 	printf("\toption\tenabled\t'0'\n\n");
 
 	return 0;
@@ -948,7 +998,28 @@
 		if (check_fs)
 			check_filesystem(pr);
 
-		err = handle_mount(pr->dev, target, pr->type, m);
+		if (strcmp(pr->type, "ntfs-3g") && strcmp(pr->type, "ntfs") && strcmp(pr->type, "exfat"))
+		{
+			err = handle_mount(pr->dev, target, pr->type, m);
+		}
+		else
+		{
+			char cmd[1024];
+			const char *param;
+
+			param = mount_flags_to_string(m->flags);
+
+			if (strcmp(pr->type, "exfat"))
+				snprintf(cmd, sizeof (cmd), "ntfs-3g -o \"%s\" %s %s >/dev/null",
+					param, pr->dev, target);
+
+			else
+				snprintf(cmd, sizeof (cmd), "mount -t exfat -o \"%s\" %s %s >/dev/null",
+					param, pr->dev, target);
+
+			err = system(cmd);
+		}
+
 		if (err)
 			ULOG_ERR("mounting %s (%s) as %s failed (%d) - %s\n",
 			         pr->dev, pr->type, target, errno, strerror(errno));
@@ -967,7 +1038,22 @@
 		if (check_fs)
 			check_filesystem(pr);
 
-		err = handle_mount(pr->dev, target, pr->type, NULL);
+		if (strcmp(pr->type, "ntfs-3g") && strcmp(pr->type, "ntfs") && strcmp(pr->type, "exfat"))
+		{
+			err = handle_mount(pr->dev, target, pr->type, NULL);
+		}
+		else
+		{
+			char cmd[1024];
+
+			if (strcmp(pr->type, "exfat"))
+				snprintf(cmd, sizeof (cmd), "ntfs-3g %s %s >/dev/null", pr->dev, target);
+			else
+				snprintf(cmd, sizeof (cmd), "mount -t exfat %s %s >/dev/null", pr->dev, target);
+
+			err = system(cmd);
+		}
+
 		if (err)
 			ULOG_ERR("mounting %s (%s) as %s failed (%d) - %s\n",
 			         pr->dev, pr->type, target, errno, strerror(errno));
