diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-09-18 18:40:09.793938006 +0800
+++ b/CMakeLists.txt	2016-09-18 18:41:52.599526829 +0800
@@ -41,6 +41,10 @@
 		libubi/ubiutils-common.c)
 INSTALL(TARGETS ubi-utils ARCHIVE DESTINATION lib)
 
+ADD_LIBRARY(fsexist STATIC
+		libfsexist/fsexist.c)
+INSTALL(TARGETS fsexist ARCHIVE DESTINATION lib)
+
 FIND_PATH(ubox_include_dir libubox/ulog.h)
 INCLUDE_DIRECTORIES(${ubox_include_dir})
 
@@ -61,9 +65,9 @@
 ADD_EXECUTABLE(block block.c)
 IF(DEFINED CMAKE_UBIFS_EXTROOT)
 	ADD_DEFINITIONS(-DUBIFS_EXTROOT)
-	TARGET_LINK_LIBRARIES(block blkid-tiny uci ubox blobmsg_json ubi-utils ${json})
+	TARGET_LINK_LIBRARIES(block blkid-tiny uci ubox blobmsg_json fsexist ubi-utils ${json})
 ELSE(DEFINED CMAKE_UBIFS_EXTROOT)
-	TARGET_LINK_LIBRARIES(block blkid-tiny uci ubox blobmsg_json ${json})
+	TARGET_LINK_LIBRARIES(block blkid-tiny uci ubox blobmsg_json fsexist ${json})
 ENDIF(DEFINED CMAKE_UBIFS_EXTROOT)
 INSTALL(TARGETS block RUNTIME DESTINATION sbin)
 
diff -Naur a/block.c b/block.c
--- a/block.c	2016-09-18 18:40:09.804938283 +0800
+++ b/block.c	2016-09-18 18:48:45.305920167 +0800
@@ -42,6 +42,8 @@
 
 #include "libblkid-tiny/libblkid-tiny.h"
 
+#include "libfsexist/fsexist.h"
+
 #ifdef UBIFS_EXTROOT
 #include "libubi/libubi.h"
 #endif
@@ -869,14 +871,20 @@
 			param = mount_flags_to_string(m->flags);
 
 			if (strcmp(pr->id->name, "exfat"))
-				snprintf(cmd, sizeof (cmd), "ntfs-3g -o \"%s,%s\" %s %s >/dev/null",
-					param, (m->options) ? (m->options) : (""), pr->dev, target);
-
+			{
+				const char fstype[]="ufsd";
+				if ( has_filesystem(fstype) ) {
+					err = mount(pr->dev, target, fstype, m->flags, (m->options) ? (m->options) : (""));
+                                } else {
+					snprintf(cmd, sizeof (cmd), "ntfs-3g -o \"%s,big_writes,%s\" %s %s >/dev/null",param, (m->options) ? (m->options) : (""), pr->dev, target);
+					err = system(cmd);
+				}
+			}
 			else
-				snprintf(cmd, sizeof (cmd), "mount -t exfat -o \"%s,%s\" %s %s >/dev/null",
-					param, (m->options) ? (m->options) : (""), pr->dev, target);
-
-			err = system(cmd);
+			{
+				snprintf(cmd, sizeof (cmd), "mount -t exfat -o \"%s,%s\" %s %s >/dev/null",param, (m->options) ? (m->options) : (""), pr->dev, target);
+				err = system(cmd);
+			}
 		}
 		if (err)
 			ULOG_ERR("mounting %s (%s) as %s failed (%d) - %s\n",
diff -Naur a/libfsexist/fsexist.c b/libfsexist/fsexist.c
--- a/libfsexist/fsexist.c	1970-01-01 08:00:00.000000000 +0800
+++ b/libfsexist/fsexist.c	2016-03-13 10:26:47.000000000 +0800
@@ -0,0 +1,45 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <errno.h>
+#include <string.h>
+#include <syslog.h>
+#include <libubox/ulog.h>
+
+#include "fsexist.h"
+
+bool has_filesystem(const char *str) {
+    FILE *fptr;
+    bool ret = false;
+    int size = strlen(str) + 3;
+    char buffer[size];
+    char *fstype = malloc(size);
+    strcpy(fstype, "\t");
+    strcat(fstype, str);
+    strcat(fstype, "\n");
+    buffer[size - 1] = fstype[size - 1];
+    fptr = fopen("/proc/filesystems", "r");
+    if (fptr == NULL)
+    {
+        ULOG_ERR("Unable to open /proc/filesystems (%s)", strerror(errno));
+        goto exit;
+    }
+    signed char c = fgetc(fptr);
+    while (c != EOF)
+    {
+        int i;
+        for (i = 0;i < size - 2;i ++)
+            buffer[i] = buffer[i + 1];
+        buffer[size - 2] = (char)c;
+        c = fgetc(fptr);
+        if (!strcmp(buffer, fstype))
+        {
+            ret = true;
+            ULOG_INFO("%s kernel space support detected!", str);
+        }
+    }
+exit:
+    fclose(fptr);
+    free(fstype);
+    return ret;
+} 
diff -Naur a/libfsexist/fsexist.h b/libfsexist/fsexist.h
--- a/libfsexist/fsexist.h	1970-01-01 08:00:00.000000000 +0800
+++ b/libfsexist/fsexist.h	2016-01-25 12:48:17.000000000 +0800
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2013 The CyanogenMod Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _FSEXIST_H
+#define _FSEXIST_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+bool has_filesystem(const char *str);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif
+
