diff -Naur a/package/feeds/packages/nfs-kernel-server/Makefile b/package/feeds/packages/nfs-kernel-server/Makefile
--- a/package/feeds/packages/nfs-kernel-server/Makefile	2016-09-12 07:41:28.000000000 +0800
+++ b/package/feeds/packages/nfs-kernel-server/Makefile	2016-09-20 10:11:10.027086189 +0800
@@ -131,8 +131,8 @@
 
 define Package/nfs-kernel-server/install
 	$(INSTALL_DIR) $(1)/etc/init.d $(1)/usr/sbin
-	$(INSTALL_DATA) ./files/nfsd.exports $(1)/etc/exports
-	$(INSTALL_BIN) ./files/nfsd.init $(1)/etc/init.d/nfsd
+	# $(INSTALL_DATA) ./files/nfsd.exports $(1)/etc/exports
+	# $(INSTALL_BIN) ./files/nfsd.init $(1)/etc/init.d/nfsd
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/utils/statd/sm-notify $(1)/usr/sbin/
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/utils/statd/statd $(1)/usr/sbin/rpc.statd
 	$(INSTALL_BIN) $(PKG_BUILD_DIR)/utils/nfsd/nfsd $(1)/usr/sbin/rpc.nfsd
